### ISSUE TITLE
obi go/rsync from a dir with spaces in it fails.

### ISSUE THREAD

**Issue by davidnixon** [Oblong/obi#51]
this came up for "~/Box Synch/someproject"
sym linking "~/Box Sync" to ~/BoxSync does not solve this.

``` diff
index ff70c8c..84370c4 100644
--- a/obi/task/task.py
+++ b/obi/task/task.py
@@ -33,7 +33,7 @@ def room_task(room_name, task_name=None):
     if not project_name:
         abort("No name key found in the project.yaml. Please specify a project name")
     env.project_name = project_name
-    env.local_project_dir = os.path.dirname(project_config)
+    env.local_project_dir = '"%s"' % os.path.dirname(project_config)
     # Abort if we cannot find room_name in rooms
     rooms = config.get("rooms", {})
     room = rooms.get(room_name, None)
```

wrapping local dir in "" fixes it.

**Comment by tomjakubowski** [comment]
That will still fail to escape other path characters (like `\` and `"`) correctly. Doesn't fabric support escaping command line arguments?

**Comment by davidnixon** [comment]
I did not try other characters mostly because I don't have a use case for them. The "Box Sync" folder is the only one I care about. I'll poke around in fabric some more.

Sent from Lotus Traveler


   Tom Jakubowski --- Re: [Oblong/obi] obi go/rsync from a dir with spaces in it fails. (#51) --- 
    From:"Tom Jakubowski" <notifications@github.com>To:"Oblong/obi" <obi@noreply.github.com>Cc:"David Nixon" <david_nixon@us.ibm.com>, "Author" <author@noreply.github.com>Date:Tue, Jun 28, 2016 6:16 PMSubject:Re: [Oblong/obi] obi go/rsync from a dir with spaces in it fails. (#51)
  
  That will still fail to escape other path characters (like \ and ") correctly. Doesn't fabric support escaping command line arguments?

**Comment by davidnixon** [comment]
@tomjakubowski @jshrake 
fabric does not have a function to escape characters but python does.
```diff
diff --git a/obi/task/task.py b/obi/task/task.py
index ff70c8c..7c9d86d 100644
--- a/obi/task/task.py
+++ b/obi/task/task.py
@@ -10,6 +10,7 @@ import fabric
 import os
 import stat
 import yaml
+import pipes
 
 from fabric.api import env  # the global env variable
 from fabric.api import (task, parallel, runs_once) #decorators
@@ -33,7 +34,7 @@ def room_task(room_name, task_name=None):
     if not project_name:
         abort("No name key found in the project.yaml. Please specify a project name")
     env.project_name = project_name
-    env.local_project_dir = os.path.dirname(project_config)
+    env.local_project_dir = pipes.quote(os.path.dirname(project_config))
     # Abort if we cannot find room_name in rooms
     rooms = config.get("rooms", {})
     room = rooms.get(room_name, None)
```
[0001-handle-special-chars-in-local-dir.patch.gz](https://github.com/Oblong/obi/files/344100/0001-handle-special-chars-in-local-dir.patch.gz)

**Comment by davidnixon** [comment]
Tested with this sub dir
```
ls -la
drwxr-xr-x 3 david_nixon david_nixon 4096 Jun 27 21:28 sub dir!@#$^&*()-_+={}[]|\:;"<>,.?
```

**Comment by tomjakubowski** [comment]
Mmm… pipes.quote is deprecated so I'm wary of using it. I was thinking more along the lines of these options to fabric's run command: http://docs.fabfile.org/en/1.11/api/core/operations.html#fabric.operations.run

But this would require surgery deeper in obi's guts than this here.

**Comment by davidnixon** [comment]
I think you're thinking too hard on this. 😉 Surgery deeper on obi would not be safer than using the deprecated pipe.quote. The run command is not really used by the rsync command anyway. The obi rsync wraps the fabric rsync and I already tried adding the shell options there. They are not accepted. 

When obi supports Python 3 then I think you could switch to show.quote but until then the pipes.quote or the original solution would handle the common use case of spaces in the directory name.

Sent from Lotus Traveler


   Tom Jakubowski --- Re: [Oblong/obi] obi go/rsync from a dir with spaces in it fails. (#51) --- 
    From:"Tom Jakubowski" <notifications@github.com>To:"Oblong/obi" <obi@noreply.github.com>Cc:"David Nixon" <david_nixon@us.ibm.com>, "Author" <author@noreply.github.com>Date:Fri, Jul 1, 2016 6:02 PMSubject:Re: [Oblong/obi] obi go/rsync from a dir with spaces in it fails. (#51)
  
  Mmm… pipes.quote is deprecated so I'm wary of using it. I was thinking more along the lines of these options to fabric's run command: http://docs.fabfile.org/en/1.11/api/core/operations.html#fabric.operations.run 
 But this would require surgery deeper in obi's guts than this here.

**Comment by sandover** [comment]
If we have a little time over the next month we'll address this.

**Comment by jshrake** [comment]
Closed by #77
